### PR TITLE
Added external-paymentjourney endpoint for external providers

### PR DESCRIPTION
**Pull request by jphillips2121**
Add a new, private, API endpoint to start an external payment session with a 3rd party provider. 

The GovPay URL is hardcoded at the moment. 

Resolves: 
CPS-142

### CODE REVIEW

File: handlers/public.go [hunk 1]
```diff
@@ -2,26 +2,8 @@ package handlers
...
-	}
-	p := &service.PaymentService{
-		DAO: m,
-	}
-
-	r.Get("/healthcheck", healthCheck).Name("get-healthcheck")
-	r.Post("/payments", p.CreatePaymentSession).Name("create-payment")
-}
-
 func healthCheck(w http.ResponseWriter, _ *http.Request) {
```
Comment (emead) [inline handlers/public.go]: This healtchCheck function could also be moved to register.go

File: routes.yaml [hunk 2]
```diff
@@ -4,3 +4,4 @@ group: api
 weight: 900	
 routes:	
    1: ^/payments
+   2: ^/private/paymentjourney
```
[the remark this answers is not in the document]
Comment (jphillips2121) [reply routes.yaml]: Spoke with architects and the correct path is ^/private/payments/XXXXX as a patch request so I have updated the path and the response code to reflect that.

File: handlers/register.go [hunk 3]
```diff
@@ -20,6 +20,7 @@ func Register(r *pat.Router, cfg config.Config) {
 
 	r.Get("/healthcheck", healthCheck).Name("get-healthcheck")
 	r.Post("/payments", p.CreatePaymentSession).Name("create-payment")
+	r.Patch("/private/payments/{payment_id}", service.CreateExternalPaymentJourney).Name("create-paymentjourney")
```
Comment (emead) [inline handlers/register.go]: Please add an extra hyphen into `create-payment-journey`

File: models/external_paysession.go [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package models
+
+// IncomingExternalPaymentJourneyRequest is the data received in the body of the incoming request
+type IncomingExternalPaymentJourneyRequest struct {
+	PaymentMethod string `json:"payment_method"`
+	Resource      string `json:"resource"`
+}
+
+// ExternalPaymentJourney returns the URL required to access external payment provider session
```
Comment (emead) [inline models/external_paysession.go]: `contains` rather than `returns`

**Pull request merged**